core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala

```diff
@@ -133,20 +168,176 @@ private[op] class OpMultiClassificationEvaluator
+      }
+    }.reduceByKey(_ + _)
+
+    labelPredictionConfidenceCountRDD.cache()
+
+    val cmByThreshold = sortedThresholds.map( threshold => {
+      val filteredRDD = labelPredictionConfidenceCountRDD.filter {
+        case ((_, _, confidence), _) => confidence >= threshold
+      }
+      constructConfusionMatrix(filteredRDD, cmClasses)
+    })
+
+    MulticlassConfMatrixMetricsByThreshold(
+      confMatrixNumClasses = $(confMatrixNumClasses),
+      confMatrixClassIndices = cmClasses,
+      confMatrixThresholds = $(confMatrixThresholds),
+      confMatrices = cmByThreshold
+    )
+  }
+
+/**
+ * function to calculate the mostly frequently mis-classified classes for each label/prediction category
+ *
+ * @param data RDD of (label, prediction)
+ * @return a MisClassificationMetrics class
+ */
+  def calculateMisClassificationMetrics(data: RDD[(Double, Double)]): MisClassificationMetrics = {
+
+    val labelPredictionCountRDD = data.map {
+      case (label, prediction) => ((label, prediction), 1L) }
+      .reduceByKey(_ + _)
+
+    val misClassificationsByLabel = labelPredictionCountRDD.map {
+      case ((label, prediction), count) => (label, (prediction, count))
+    }.groupByKey()
+      .map { case (label, predictionCountsIter) => {
+        val misClassificationCtMap = predictionCountsIter.toSeq
+          .filter { case (pred, _) => pred != label }
+          .sortBy(-_._2)
+          .take($(confMatrixMinSupport)).toMap
+
+        val labelCount = predictionCountsIter.map(_._2).reduce(_ + _)
+        val correctCount = predictionCountsIter.filter {case (pred, _) => pred == label }
+          .map(_._2)
+          .reduceOption(_ + _).getOrElse(0L)
+
+        MisClassificationsPerCategory(
+          category = label,
+          totalCount = labelCount,
+          correctCount = correctCount,
+          misClassifications = misClassificationCtMap
+        )
+      }
+    }.collect()
+     .sortBy(-_.totalCount).toSeq
+
+    val misClassificationsByPrediction = labelPredictionCountRDD.map {
+      case ((label, prediction), count) => (prediction, (label, count))
+    }.groupByKey()
+      .map { case (prediction, labelCountsIter) => {
+        val sortedMisclassificationCt = labelCountsIter.toSeq
+          .filter { case (label, _) => label != prediction }
+          .sortBy(-_._2)
+          .take($(confMatrixMinSupport)).toMap
+
+        val predictionCount = labelCountsIter.map(_._2).reduce(_ + _)
+        val correctCount = labelCountsIter.filter {case (label, _) => label == prediction }
+            .map(_._2)
+            .reduceOption(_ + _).getOrElse(0L)
+
+        MisClassificationsPerCategory(
+          category = prediction,
+          totalCount = predictionCount,
+          correctCount = correctCount,
+          misClassifications = sortedMisclassificationCt
+        )
+      }
+    }.collect()
+      .sortBy(-_.totalCount).toSeq
```

Comment:
can we reverse sortBy and collect to give Spark more work? 

toSeq is redundant